core/src/main/java/org/axonframework/eventhandling/AbstractEventProcessor.java

```diff
@@ -51,7 +52,7 @@
 public abstract class AbstractEventProcessor implements EventProcessor {
 
     private final Logger logger = LoggerFactory.getLogger(getClass());
-    private final Set<MessageHandlerInterceptor<? super EventMessage<?>>> interceptors = new CopyOnWriteArraySet<>();
+    private final List<MessageHandlerInterceptor<? super EventMessage<?>>> interceptors = new ArrayList<>();
```

Comment:
Please use CopyOnWriteArrayList for thread safety. It's possible to (un)register interceptors at runtime.